core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1686,12 +1847,208 @@ public void warmUp() {
+                }
+            }
+
+            // detached could have had more columns than we need
+            for (int i = columnCount, limit = detachedMetadata.getColumnCount(); i < limit; i++) {
+                String columnName = detachedMetadata.getColumnName(i);
+                detachedDeleteExtraColNames.add(columnName);
+            }
+
+            // load/check _dcv, updating local column tops
+            // set current _dcv to where the partition was
+            other2.parent().concat(DETACHED_COLUMN_VERSION_FILE_NAME).$();
+            if (!ff.exists(other2)) {
+                // Backups and older versions of detached partitions will not have _cv
+                LOG.error().$("detached _cv not found, skipping check [path=").$(other2).I$();
+            } else {
+                if (detachedColumnVersionReader == null) {
+                    detachedColumnVersionReader = new ColumnVersionReader();
+                }
+                detachedColumnVersionReader.ofRO(ff, other2);
+                detachedColumnVersionReader.readSafe(MillisecondClockImpl.INSTANCE, Long.MAX_VALUE);
+
+                // set column tops for missing columns
+                ObjList<CharSequence> keys = detachedAddMissingColNames.keys();
+                for (int i = 0, limit = keys.size(); i < limit; i++) {
+                    CharSequence columnName = keys.get(i);
+                    long columnTop;
+                    int colIdx = detachedAddMissingColNames.get(columnName);
+                    if (colIdx < 0) { // column was added, we have no data
+                        colIdx = metadata.getColumnIndex(columnName);
+                        columnTop = partitionSize;
+                    } else {
+                        columnTop = -1L;
+                        for (int j = colIdx, limit2 = detachedMetadata.getColumnCount(); j < limit2; j++) {
+                            if (j == metadata.getColumnIndex(columnName) && metadata.getColumnType(j) == detachedMetadata.getColumnType(j)) {
+                                CharSequence dColumnName = detachedMetadata.getColumnName(j);
+                                throw CairoException.instance(CairoException.METADATA_VALIDATION)
+                                        .put("Potential rename of column ")
+                                        .put(columnName).put(" to ").put(dColumnName)
+                                        .put(", this is not supported, seek manual intervention");
+                            }
+                        }
+                    }
+                    columnVersionWriter.upsertColumnTop(partitionTimestamp, colIdx, columnTop);
+                }
+
+                // override column tops for existing columns
+                columnVersionWriter.upsertPartition(partitionTimestamp, detachedColumnVersionReader);
+
+                // delete extra columns (present in .attachable - the table does not track/need them)
```

Comment:
Let's not delete extra column files, some users will want to attach / detach same partition multiple times without data to be changed.

We can delete index files, they can be restored